uwblueprint/jump-math
# fix: admin verification issues

## Notion ticket link

[Fix Admin Signup Confirmation](https://www.notion.so/uwblueprintexecs/Fix-Admin-Signup-Confirmation-0fe47b60ea664d9a9dddba8caa893a2c)



## Implementation description
- Remove dangerous `RESET_PASSWORD_CODE` mutation
- Refactor backend to support logic [here](https://www.notion.so/uwblueprintexecs/Fix-Admin-Signup-Confirmation-0fe47b60ea664d9a9dddba8caa893a2c)
- Refactor front-end to fix refresh bug
- Fix logic error in forgot-password component

**Note:** You _could_ consider [reviewing by commit](https://github.com/uwblueprint/jump-math/pull/594/commits/8316d7e1b3efe32301c4badec1d707119b2d4e74?w=1) to reduce cognitive load, and I tried my best to break things up. However, a lot of the changes here are interrelated, so this is a judgement call.



## Steps to test
This one is a doozy...

Forgot password:
1. For an account you own, click "Forgot Password," fill out the form (including testing validation logic), and submit.
2. Wait until you receive your email and click the link.
3. After the page finishes loading, hit refresh and wait for the page to fully load with no errors.
4. Fill out the reset-password form.
5. Verify you can log in with your new password.
6. Re-click the link in your email and verify you see an error page now.

Verify account (teacher):
1. Create a new teacher account.
2. Wait until you receive your email and click the link.
3. After the page finishes loading, hit refresh and wait for the page to fully load with no errors.
5. Verify you can log in with your password.
6. Re-click the link in your email and verify you see an error page now.

Verify account (admin):
1. Invite your email to be a new admin account.
2. Wait until you receive your email and click the link.
3. After the page finishes loading, hit refresh and wait for the page to fully load with no errors.
4. Fill out the reset-password form.
5. Verify you can log in with your new password.
6. Re-click the link in your email and verify you see an error page now.



## What should reviewers focus on?
* See my comment on Slack: wondering about whether we want to add "resend verification email" to the verification error pages and a link back to the initial "Forgot password" page on the forgot password error page.


## Checklist


[test](https://github.com/uwblueprint/jump-math/pull/594/files?w=1)
